.github/guidelines/CODING_GUIDELINES.md

```diff
@@ -0,0 +1,77 @@
+# MetaMask Coding Guidelines
+
+### 1. New Code Should be Typescript
+- New components and utilities should be written in Typescript and enforce typing.
+- Existing code should be refactored into Typescript where time allows. If you are replacing a component, use Typescript.
+
```

Comment:
We could reference the contributor docs TypeScript best practices here. Something like
```suggestion
Follow contributor doc [TypeScript Guidelines](https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md)

```

Reply:
Added in this [commit](https://github.com/MetaMask/metamask-extension/pull/20651/commits/147f556dfd97dd3f426d9d9f8bc6a01a2c1f27ff).